### PR TITLE
Chen development

### PR DESCRIPTION
**Pull request by chenraff**
Week 2 starter project - China portfolio page.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -12,17 +12,23 @@
...
-  greetingContainer.innerText = greeting;
+//      
+function advanceSlide(advanceAmount){
+    showSlide(currSlideIndex += advanceAmount);
 }
+
+//
+function showSlide(slideIndex){
+    slides.forEach(slide => slide.style.display = "none");
+    slideIndex = mod(slideIndex, slides.length);
```
Comment (mayascha) [inline portfolio/src/main/webapp/script.js:64]: Cool implementation!

File: portfolio/src/main/webapp/style.css [hunk 2]
```diff
@@ -4,6 +4,48 @@
...
+.prev, .next {
+    cursor: pointer;
+    position: absolute;
+    top: 50%;
+    margin-top: -25px;
+    color: whitesmoke;
+    font-size: 40px;
+    font-weight: bold;
+    padding: 18px;
+
```
Comment (mayascha) [inline portfolio/src/main/webapp/style.css:70]: Redundant empty line

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (mayascha) [inline portfolio/src/main/webapp/script.js:64]: js coding style - (slide) in brackets

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -12,17 +12,23 @@
...
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+//      
+function advanceSlide(advanceAmount){
+    showSlide(currSlideIndex += advanceAmount);
 }
+
+//
+function showSlide(slideIndex){
+    slides.forEach(slide => slide.style.display = "none");
```
Comment (mayascha) [inline portfolio/src/main/webapp/script.js]: In JS we usually use ' and not "
In addition, I think you can define a constant for 'none'. let see what Eyal thinks

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -12,17 +12,23 @@
...
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+var currSlideIndex = 0;
+var slides = document.querySelectorAll(".slide");
+showSlide(currSlideIndex);
```
Comment (mayascha) [inline portfolio/src/main/webapp/script.js:49]: JS coding style - better to use either let or const, nor var

**Review by mayascha — COMMENTED** [review]
Great job Chen. I've wrote some nits, let's wait for review from Eyal as well :)

**Review by mayascha — APPROVED** [review]

**Review by mayascha — APPROVED** [review]

**Review by mayascha — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (ewebdev) [reply portfolio/src/main/webapp/script.js:49]: +1 - use "const" when possible, "let" otherwise.

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -1,18 +1,92 @@
...
   <body>
     <div id="content">
-      <h1>My Portfolio</h1>
-      <p>This is my portfolio.</p>
-      <p>Click here to get a random message:</p>
-      <button onclick="addRandomGreeting()">Hello</button>
-      <div id="greeting-container"></div>
+      <h1>Discover China 🇨🇳</h1>
+      <h2>你好! (Nǐ hǎo!)</h2>
+      <p><b>China</b> (中国), officially the People's Republic of China, is a country in East Asia. </p>
```
Comment (ewebdev) [inline portfolio/src/main/webapp/index.html]: Prefer using <strong> over <b> tags (and <em> over <i>)
```suggestion
      <p>
        <strong>China</strong> (中国), officially the People's Republic of China, is a country in East Asia.
      </p>
```
https://stackoverflow.com/questions/271743/whats-the-difference-between-b-and-strong-i-and-em
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/strong

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -1,18 +1,92 @@
...
+              </tr>
+              <tr>
+                  <td><b>Currency</b></td>
+                  <td>Renminbi (yuan; ¥; CNY)</td>
+              </tr>
+              <tr>
+                  <td><b>Time zone</b></td>
+                  <td>UTC+8 (China Standard Time)</td>
+              </tr>
+
```
Comment (ewebdev) [inline portfolio/src/main/webapp/index.html]: Remove newline

File: portfolio/src/main/webapp/index.html [hunk 7]
```diff
@@ -1,18 +1,92 @@
...
+        <li>2,200 years ago, the game that later turned into football (soccer) was invented in China.</li>      
+      </ol>
+      <p>For further information please visit <a href="https://en.wikipedia.org/wiki/China" target="_blank">China's wikipedia page</a></p>
+      <h3>Now that we've covered the basics , let's look at some pictures from China:</h3>
+    </div>
+    
+    <!-- China's picture slideshow -->
+    <div class="slideshow-container">
+        <div class="slide fade">
+            <img src="Images/China-1.jpg" style="width:100%">
```
Comment (ewebdev) [inline portfolio/src/main/webapp/index.html]: Move all inline style declarations to the CSS file 
https://google.github.io/styleguide/htmlcssguide.html#Separation_of_Concerns
```suggestion
            <img src="Images/China-1.jpg">
```

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -12,17 +12,23 @@
...
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+// Modulus operator (for positive and negative numbers)
+function mod(a,b) {
+    return ((a % b) + b) % b; 
+} 
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+//
```
Comment (ewebdev) [inline portfolio/src/main/webapp/script.js]: Remove empty comment

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -12,17 +12,23 @@
...
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+//      
+function advanceSlide(advanceAmount){
+    showSlide(currSlideIndex += advanceAmount);
 }
+
+//
```
Comment (ewebdev) [inline portfolio/src/main/webapp/script.js]: Remove empty comment

File: portfolio/src/main/webapp/index.html [hunk 10]
```diff
@@ -1,18 +1,50 @@
...
+      <p>Let's Look at some cool pictures from China:</p>
+    </div>
+
+    <div class="slideshow-container">
+
+        <div class="slide fade">
+            <img src="Images/China-1.jpg" style="width:100%">
+        </div>
+        <div class="slide fade">
+            <img src="Images/China-2.jpg" style="width:100%">
```
Comment (ewebdev) [inline portfolio/src/main/webapp/index.html]: Same here in and other images 
```suggestion
            <img src="Images/China-2.jpg">
```

File: portfolio/src/main/webapp/index.html [hunk 11]
```diff
@@ -1,18 +1,50 @@
...
+        <div class="slide fade">
+            <img src="Images/China-6.jpg" style="width:100%">
+        </div>
+        <div class="slide fade">
+            <img src="Images/China-7.jpg" style="width:100%">
+        </div>
+        <div class="slide fade">
+            <img src="Images/China-8.jpg" style="width:100%">
+        </div>
+
```
Comment (ewebdev) [inline portfolio/src/main/webapp/index.html]: Remove 1 newline

File: portfolio/src/main/webapp/style.css [hunk 12]
```diff
@@ -4,6 +4,48 @@
...
+  margin: auto;
+  align-items: center;
+}
+
+/* Hide slides at first */
+.slide {
+    display: none;
+}
+
+.prev, .next {
```
Comment (ewebdev) [inline portfolio/src/main/webapp/style.css]: Separate selectors and declarations by new lines.
https://google.github.io/styleguide/htmlcssguide.html#Selector_and_Declaration_Separation

File: portfolio/src/main/webapp/style.css [hunk 13]
```diff
@@ -4,6 +4,48 @@
...
+
+.prev {
+    left: 2px;
+}
+
+.next {
+    right: 2px;
+}
+
+.prev: hover, .next:hover {
```
Comment (ewebdev) [inline portfolio/src/main/webapp/style.css]: Separate selectors and declarations by new lines.
https://google.github.io/styleguide/htmlcssguide.html#Selector_and_Declaration_Separation

File: portfolio/src/main/webapp/style.css [hunk 14]
```diff
@@ -4,6 +4,48 @@
...
+    background-color = rgba(0,0,0, 0.7);
+    color = whitesmoke;
```
Comment (ewebdev) [inline portfolio/src/main/webapp/style.css]:
```suggestion
    background-color: rgba(0,0,0, .7);
    colo: whitesmoke;
```

File: portfolio/src/main/webapp/style.css [hunk 15]
```diff
@@ -4,6 +4,48 @@
...
+    margin-top: -25px;
+    color: whitesmoke;
+    font-size: 40px;
+    font-weight: bold;
+    padding: 18px;
+
+}
+
+.prev {
+    left: 2px;
```
Comment (ewebdev) [inline portfolio/src/main/webapp/style.css]: Indent by 2 spaces at a time.
https://google.github.io/styleguide/htmlcssguide.html#Indentation

**Review by ewebdev — COMMENTED** [review]
Very nice! congrats on your first pull request!

**Review by ewebdev — COMMENTED** [review]
A few more things:
1. Always use lowercase file/folder names 
  https://google.github.io/styleguide/jsguide.html#file-name
2. Consider using "auto format" in your IDE to avoid formatting, it might be needed to be configured for compatibility with the Google's style guide (e.g. indentation of 2 whitespace, newlines etc.).

File: portfolio/src/main/webapp/style.css (hunk 2 above)
Comment (chenraff) [reply portfolio/src/main/webapp/style.css:70]: removed

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (chenraff) [reply portfolio/src/main/webapp/script.js:64]: done :)

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (chenraff) [reply portfolio/src/main/webapp/script.js]: done :)

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (chenraff) [reply portfolio/src/main/webapp/script.js:49]: done :)

File: portfolio/src/main/webapp/index.html (hunk 5 above)
Comment (chenraff) [reply portfolio/src/main/webapp/index.html]: done :)  (the explanation on stack overflow in regards to accessibility was really cool!)

File: portfolio/src/main/webapp/index.html (hunk 6 above)
Comment (chenraff) [reply portfolio/src/main/webapp/index.html]: done :)
Is there any rule of thumb on where you should/shouldn't add new line on HTML file? I didn't find anything about this in the style guide

File: portfolio/src/main/webapp/index.html (hunk 7 above)
Comment (chenraff) [reply portfolio/src/main/webapp/index.html]: done :)

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (chenraff) [reply portfolio/src/main/webapp/script.js]: done :)

File: portfolio/src/main/webapp/index.html (hunk 11 above)
Comment (chenraff) [reply portfolio/src/main/webapp/index.html]: done :)

File: portfolio/src/main/webapp/style.css (hunk 12 above)
Comment (chenraff) [reply portfolio/src/main/webapp/style.css]: done :)

File: portfolio/src/main/webapp/style.css (hunk 13 above)
Comment (chenraff) [reply portfolio/src/main/webapp/style.css]: done :)

File: portfolio/src/main/webapp/style.css (hunk 14 above)
Comment (chenraff) [reply portfolio/src/main/webapp/style.css]: done (I removed this style in one of the previous versions)

File: portfolio/src/main/webapp/style.css (hunk 15 above)
Comment (chenraff) [reply portfolio/src/main/webapp/style.css]: done :)

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (chenraff) [reply portfolio/src/main/webapp/script.js:64]: Thanks!

File: portfolio/src/main/webapp/index.html (hunk 6 above)
Comment (ewebdev) [reply portfolio/src/main/webapp/index.html]: Basically only add new lines when it improves readability and as few as possible :)

**Pull request merged**